### PR TITLE
Hide the `ci_support` directory

### PR DESCRIPTION
**Pull request by jakirkham**
Move the `ci_support` directory to `.ci_support` to keep it hidden.

### CODE REVIEW

**Comment by jakirkham** [comment]
Not sure how we feel about this. So have labeled it RFC for now. Feedback welcome.

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -548,12 +548,12 @@ def _load_forge_config(forge_dir, variant_config_files):
...
-        os.path.join('ci_support', 'upload_or_check_non_existence.py'),
+        os.path.join('.ci_support', 'upload_or_check_non_existence.py'),
         'circle.yml',
         'appveyor.yml',
-        os.path.join('ci_support', 'checkout_merge_commit.sh'),
-        os.path.join('ci_support', 'fast_finish_ci_pr_build.sh'),
-        os.path.join('ci_support', 'run_docker_build.sh'),
+        os.path.join('.ci_support', 'checkout_merge_commit.sh'),
+        os.path.join('.ci_support', 'fast_finish_ci_pr_build.sh'),
+        os.path.join('.ci_support', 'run_docker_build.sh'),
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py]: This should be reverted

**Comment by isuruf** [comment]
`ci_support` is now a folder with only `matrix` folder in it. Do we want to remove move the contents of `matrix` one level up?

**Comment by jakirkham** [comment]
That sounds like a good idea. Another thought is we could move these into respective directories for each CI. No strong feelings about either option from me.

**Comment by isuruf** [comment]
> Another thought is we could move these into respective directories for each CI. No strong feelings about either option from me.

Having them all in one place is better, I think.

**Comment by jakirkham** [comment]
Suppose we could rename it to something like `.ci_matrix` as well.

**Comment by jakirkham** [comment]
Anything else, @isuruf?

**Pull request merged**